Add more lambda expressions in data-access code examples

maxxedev opened the pull request:
add more java8 lambda expressions in documentation for data-access

maxxedev committed:
more java8 lambda expressions in code examples

maxxedev commented on src/docs/asciidoc/data-access.adoc:
```diff
@@ -5567,16 +5558,14 @@ declared type `ITEM_TYPE`:
 			// ...
 			declareParameter(new SqlOutParameter("item", OracleTypes.STRUCT, "ITEM_TYPE",
-				new SqlReturnType() {
-					public Object getTypeValue(CallableStatement cs, int colIndx, int sqlType, String 	) throws SQLException {
```
missing param name in `String typeName`

sbrannen replied:
That's true, but this is in the documentation that is being replaced. Thus we can ignore that for now.

sbrannen commented on src/docs/asciidoc/data-access.adoc:
```diff
@@ -2730,17 +2730,15 @@ The following query finds and populates a single domain object:
 [source,java,indent=0,subs="verbatim,quotes",role="primary"]
 .Java
 ----
+	RowMapper<Actor> actorRowMapper = (resultRest, rowNum) -> {
```
`resultRest` --> `resultSet`

Please make that change consistently in this PR.

sbrannen replied:
Please inline the lambda expression as was previously the case with the anonymous inner class, since that then aligns with the explanatory text in the following paragraph.

sbrannen commented on src/docs/asciidoc/data-access.adoc:
```diff
@@ -2757,16 +2755,15 @@ The following query finds and populates a number of domain objects:
 [source,java,indent=0,subs="verbatim,quotes",role="primary"]
 .Java
 ----
+	RowMapper<Actor> actorRowMapper = (resultRest, rowNum) -> {
```
Please inline this lambda expression as well. (see previous comment)

sbrannen commented on src/docs/asciidoc/data-access.adoc:
```diff
@@ -2777,26 +2774,25 @@ The following query finds and populates a number of domain objects:
-	private static final class ActorMapper implements RowMapper<Actor> {
-
-		public Actor mapRow(ResultSet rs, int rowNum) throws SQLException {
+	private RowMapper<Actor> getActorMapper() {
```
The `RowMapper` implementation is stateless. Therefore, please store a single reference to it in a `final` field and remove the `getActorMapper()` method.

maxxedev replied:
Ok, stored as instance field `actorRowMapper`. Let know static field `ACTOR_ROW_MAPPER` is preferred

sbrannen commented on src/docs/asciidoc/data-access.adoc:
```diff
@@ -3562,15 +3558,12 @@ on Oracle but may not work on other platforms:
+		ps.setString(1, name);
+		return ps;
+	};
+	jdbcTemplate.update(statementCreator, keyHolder);
```
Please inline the lambda expression.

maxxedev committed:
more polishing on variable name and style.

fix typo on resultSet.
inline lambda expressions to be consistent with explanatory text.
store row mapper as a final field.

maxxedev commented:
@sbrannen anything else?

sbrannen commented:
> @sbrannen anything else?

Thanks for making the changes. I think that looks pretty good now, and I'll review it locally before merging.

sbrannen commented:
This has been merged into `master` in 9b827283f1502c7bf7d517597cad47ce26944625 and refined in 7dea2686b81509f7d42901da5f0d68ae2e69bee9.

Thanks

sbrannen commented:
As a side note, the last Kotlin example in [this section](https://github.com/spring-projects/spring-framework/blob/master/src/docs/asciidoc/data-access.adoc#querying-select) still uses a local class instead of a lambda expression.

I'll see if we can that addressed as well.

maxxedev commented:
thanks for the review!

the [newer example](https://github.com/spring-projects/spring-framework/commit/7dea2686b81509f7d42901da5f0d68ae2e69bee9#r37967323) may not compile though.

sbrannen commented:
> thanks for the review!

You're welcome.

> the [newer example](https://github.com/spring-projects/spring-framework/commit/7dea2686b81509f7d42901da5f0d68ae2e69bee9#r37967323) may not compile though.

Fixed in b52136dfa5fcda00a6a5073470c6751ce96e1953.

Thanks again for pointing that out! 👍
